File: leetcode/537. Complex Number Multiplication/537.js
// https://helloacm.com/the-complex-number-multiplication-function/
// https://leetcode.com/problems/complex-number-multiplication/

/**
 * @param {string} a
 * @param {string} b
 * @return {string}
 */
var complexNumberMultiply = function(a, b) {
    const process = (x) => {
        x = x.split('+');
        const a = x[0];
        const b = x[1].slice(0, -1);
        return [a, b];
    }
    const aa = process(a);
    const bb = process(b);
    const xx = aa[0] * bb[0] - aa[1] * bb[1];
    const yy = aa[0] * bb[1] + aa[1] * bb[0];
    return `${xx}+${yy}i`;
};

File: leetcode/537. Complex Number Multiplication/537.cpp
// https://helloacm.com/the-complex-number-multiplication-function/
// https://leetcode.com/problems/complex-number-multiplication/submissions/

class Solution {
public:
    string complexNumberMultiply(string a, string b) {
        int a1, a2, b1, b2;
        sscanf(a.c_str(), "%d+%di", &a1, &a2);
        sscanf(b.c_str(), "%d+%di", &b1, &b2);
        int xx = a1 * b1 - a2 * b2;
        int yy = a1 * b2 + a2 * b1;
        return std::to_string(xx) + "+" + std::to_string(yy) + "i";
    }
};
